lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,715 @@
+        if buf.is_empty() {
+            return Ok(None);
+        }
+
+        Ok(self
+            .bytes_decoder
+            .decode_eof(buf)?
+            .and_then(|frame| self.decode_message(frame).transpose())
+            .transpose()?)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use bytes::{BufMut, BytesMut};
+    use rstest::{fixture, rstest};
+    use tracing_test::traced_test;
+
+    fn create_chunk(
+        message_id: u64,
+        sequence_number: u8,
+        total_chunks: u8,
+        payload: &str,
+    ) -> BytesMut {
+        let mut chunk = BytesMut::new();
+        chunk.put_slice(&GELF_MAGIC);
+        chunk.put_u64(message_id);
+        chunk.put_u8(sequence_number);
+        chunk.put_u8(total_chunks);
+        chunk.extend_from_slice(payload.as_bytes());
+        chunk
+    }
+
+    #[fixture]
+    fn unchunked_message() -> (BytesMut, String) {
+        let payload = "foo";
+        (BytesMut::from(payload), payload.to_string())
+    }
+
+    #[fixture]
+    fn two_chunks_message() -> ([BytesMut; 2], String) {
+        let message_id = 1u64;
+        let total_chunks = 2u8;
+
+        let first_sequence_number = 0u8;
+        let first_payload = "foo";
+        let first_chunk = create_chunk(
+            message_id,
+            first_sequence_number,
+            total_chunks,
+            first_payload,
+        );
+
+        let second_sequence_number = 1u8;
+        let second_payload = "bar";
+        let second_chunk = create_chunk(
+            message_id,
+            second_sequence_number,
+            total_chunks,
+            second_payload,
+        );
+
+        (
+            [first_chunk, second_chunk],
+            format!("{first_payload}{second_payload}"),
+        )
+    }
+
+    #[fixture]
+    fn three_chunks_message() -> ([BytesMut; 3], String) {
+        let message_id = 2u64;
+        let total_chunks = 3u8;
+
+        let first_sequence_number = 0u8;
+        let first_payload = "foo";
+        let first_chunk = create_chunk(
+            message_id,
+            first_sequence_number,
+            total_chunks,
+            first_payload,
+        );
+
+        let second_sequence_number = 1u8;
+        let second_payload = "bar";
+        let second_chunk = create_chunk(
+            message_id,
+            second_sequence_number,
+            total_chunks,
+            second_payload,
+        );
+
+        let third_sequence_number = 2u8;
+        let third_payload = "baz";
+        let third_chunk = create_chunk(
+            message_id,
+            third_sequence_number,
+            total_chunks,
+            third_payload,
+        );
+
+        (
+            [first_chunk, second_chunk, third_chunk],
+            format!("{first_payload}{second_payload}{third_payload}"),
+        )
+    }
+
+    fn downcast_framing_error(error: &Box<dyn FramingError>) -> &ChunkedGelfDecoderError {
```

Comment:
Had to do this hack in order to match a `Box<dyn FramingError` to a specific `ChunkedGelfDecoderError` enum variant